src/queue/consumers/oneidentity/consumerCallbacks/oneIdentityIntegrationHandler.ts

```diff
@@ -0,0 +1,147 @@
+import { logger } from '@user-office-software/duo-logger';
+
+import { Event } from '../../../../models/Event';
+import { ProposalMessageData } from '../../../../models/ProposalMessage';
+import { ProposalUser } from '../../scicat/scicatProposal/dto';
+import {
+  ESSOneIdentity,
+  PersonHasESETValues,
+  UID_ESet,
+  UID_Person,
+  UserPersonConnection,
+} from '../utils/ESSOneIdentity';
+
+export async function oneIdentityIntegrationHandler(
+  message: ProposalMessageData,
+  type: Event
+): Promise<void> {
+  // Create a new ESSOneIdentity instance and log in
+  const oneIdentity = new ESSOneIdentity();
+  await oneIdentity.login();
+
+  try {
+    const uidESet = await getUIDESetFromOneIdentity(oneIdentity, message, type);
+
+    if (uidESet) {
+      await handleConnectionsBetweenProposalAndPersons(
+        oneIdentity,
+        uidESet,
+        message
+      );
+    }
+  } finally {
+    await oneIdentity.logout();
+  }
+}
+
+// Method to get UID_ESet from One Identity, create it if it does not exist
+async function getUIDESetFromOneIdentity(
+  oneIdentity: ESSOneIdentity,
+  message: ProposalMessageData,
+  type: Event
+): Promise<UID_ESet | undefined> {
+  let uidESet: UID_ESet | undefined = await oneIdentity.getProposal(message);
+
+  logger.logInfo('Proposal in One Identity', { uidESet });
+
+  if (type === Event.PROPOSAL_UPDATED && !uidESet) {
+    // Proposal does not exist in One Identity
+    return;
+  }
+
+  if (!uidESet) {
+    // Create proposal in One Identity if it does not exist
+    uidESet = await oneIdentity.createProposal(message);
+
+    logger.logInfo('UID_ESet created', { uidESet });
+
+    if (!uidESet) {
+      throw new Error('Proposal creation failed in ESS One Identity');
+    }
+  }
+
+  return uidESet;
+}
+
+async function handleConnectionsBetweenProposalAndPersons(
```

Comment:
I think everywhere you have used `persons` should be replaced with `people`.

Reply:
I would use the word 'Person(s)' because it corresponds to the table name in 1IM.